pandas/core/groupby/groupby.py

```diff
@@ -2890,12 +2891,21 @@ def size(self) -> DataFrame | Series:
         Freq: MS, dtype: int64
         """
         result = self.grouper.size()
+        result_dtype: str | np.dtype = result.dtype
+        if isinstance(self.obj, Series):
+            if isinstance(self.obj.array, ArrowExtensionArray):
+                result_dtype = "int64[pyarrow]"
+            elif isinstance(self.obj.array, BaseMaskedArray):
+                result_dtype = "Int64"
+        # TODO: For DataFrames what if columns are mixed arrow/numpy/masked?
```

Comment:
Currently adding Series np + masked or np + arrow gives masked and arrow respectively (regardless of order). I think we could follow the same here. However masked + arrow raises whereas I would think we need to make a choice here. Not sure if this has been discussed much in other contexts?

Reply:
Yeah I don't think this has been discussed (but has been looming). I would prefer to punt on this for now until it's more formally discussed

Reply:
im fine with punting.  id also point out that `__pandas_priority__` was implemented partly with this in mind